webpages/settings/style.css

```diff
@@ -278,11 +278,13 @@ body.iframe .settings-column > .addon-setting {
   flex-direction: column;
   align-items: flex-start;
 }
-body.iframe .setting-label {
-  width: auto;
-}
 body.iframe .setting-label-container {
   margin-bottom: 5px;
+  width: auto;
+  max-width: calc(100% - 66px); /* prevent setting description tooltip from overflowing */
+}
+body.iframe .setting-label-container:not(:has(.tooltip)) {
+  max-width: none;
 }
```

Comment:
In browsers that don't support `:has`, the `max-width` will be applied to all labels, which shouldn't cause any issues.